tools/template_based_query_generation/src/main/java/token/QueryRegex.java

```diff
@@ -0,0 +1,115 @@
+package token;
+
+import parser.Utils;
+import query.SkeletonPiece;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+public class QueryRegex {
```

Comment:
Looks like this class only contains one function, so probably it makes more sense to move this function to Query class?